processor/configuration/src/test/java/com/redhat/service/smartevents/processor/sources/aws/AwsS3SourceConnectorTest.java

```diff
@@ -0,0 +1,103 @@
+    private static final String TEST_TOPIC_NAME = "test-topic-name";
+    private static final String TEST_PREFIX = "test-prefix";
+
+    private static final String EXPECTED_PAYLOAD_JSON = "{" +
+            "   \"aws_bucket_name_or_arn\":\"" + TEST_BUCKET + "\"," +
+            "   \"aws_region\":\"" + TEST_REGION + "\"," +
+            "   \"aws_access_key\":\"" + TEST_ACCESS_KEY + "\"," +
+            "   \"aws_secret_key\":\"" + TEST_SECRET_KEY + "\"," +
+            "   \"aws_prefix\":\"" + TEST_PREFIX + "\"," +
+            "   \"aws_ignore_body\":" + TEST_IGNORE_BODY + "," +
+            "   \"aws_delete_after_read\":" + TEST_DELETE_AFTER_READ + "," +
+            "   \"kafka_topic\":\"" + TEST_TOPIC_NAME + "\"," +
+            "   \"processors\": [" +
+            "       {" +
+            "           \"log\": {" +
+            "               \"multiLine\":true," +
+            "               \"showHeaders\":true" +
+            "        }" +
+            "     }" +
+            "   ]" +
+            "}";
+
+    @Inject
+    AwsS3SourceConnector connector;
+
+    @Inject
+    ObjectMapper mapper;
+
+    @Test
+    void testConnectorType() {
+        assertThat(connector.getConnectorTypeId()).isEqualTo(AwsS3SourceConnector.CONNECTOR_TYPE_ID);
+    }
+
+    @Test
+    void testConnectorPayload() throws JsonProcessingException {
+        JsonNode expectedPayload = mapper.readTree(EXPECTED_PAYLOAD_JSON);
+
+        Source source = new Source();
+        source.setType(AwsS3SourceConnector.TYPE);
+        source.setParameters(Map.of(AwsS3Source.BUCKET_NAME_OR_ARN_PARAMETER, TEST_BUCKET,
+                AwsS3Source.REGION_PARAMETER, TEST_REGION,
+                AwsS3Source.ACCESS_KEY_PARAMETER, TEST_ACCESS_KEY,
+                AwsS3Source.SECRET_KEY_PARAMETER, TEST_SECRET_KEY,
+                AwsS3Source.PREFIX, TEST_PREFIX,
+                AwsS3Source.IGNORE_BODY_PARAMETER, TEST_IGNORE_BODY.toString(),
+                AwsS3Source.DELETE_AFTER_READ_PARAMETER, TEST_DELETE_AFTER_READ.toString()));
+
+        JsonNode payload = connector.connectorPayload(source, TEST_TOPIC_NAME);
+
+        assertThat(payload).isEqualTo(expectedPayload);
+    }
+
+    @Test
+    void testEmptyPrefix() {
+        Source source = new Source();
+        source.setType(AwsS3SourceConnector.TYPE);
+        source.setParameters(Map.of(AwsS3Source.PREFIX, ""));
+
+        JsonNode payload = connector.connectorPayload(source, TEST_TOPIC_NAME);
+
+        assertThat(payload.asText()).doesNotContain("prefix");
+    }
+
+    @Test
+    void testNullPrefix() {
+        Source source = new Source();
+        source.setType(AwsS3SourceConnector.TYPE);
+        Map<String, String> values = new HashMap<>();
+        values.put(AwsS3Source.PREFIX, null);
+        source.setParameters(values);
+
+        JsonNode payload = connector.connectorPayload(source, TEST_TOPIC_NAME);
+
+        assertThat(payload.asText()).doesNotContain("prefix");
+    }
```

Comment:
For these two test cases, I am actually not sure what we want to check with `doesNotContain`.

is "prefix" just some substring to be sure it is not present at all. Or was our intention to assert one of  bellow is not present:
- TEST_PREFIX
- AwsS3Source.PREFIX
- "aws_prefix"